json-logic/README.md

```diff
@@ -0,0 +1,62 @@
+# CovQuestions Draft
+
+This is the draft of a custom questionaire format for the CovQuestions project.
+
+## Requirements
+
+* Relatively easy to edit
+* Easy to use and implement
+* As platform-independend as possible
+* Describe questions, results, and result-computation logic
+* Questions might be shown or not depending in intermediate results
+* Support for multiple result categories, with their own results and computations
+
+## Considerations
+
+### Result Format
+
+Two possible options: 
+1. "Flat" list of results with a expressione evaluating to a boolean
+2. Result categories with an expression that evaluate to a result id with a list of results per category.
+
+We favor option 2 - with option 1 it is easy to accidentially show two results from the same category. This is impossible with 2 by design at the price of one additional layer of nesting.
+
+### Evaluation
+
+Two possible options: 
+1. Put all variables and questions into the same array and process them in order.
+2. Seperate variables and questions, process all questions in order and (re-)evaluate all variables after each update.
+
+We favor option 2 - it gives the user one less thing to worry about.
+
+#### Structuring Expressions
+
+JSON Logic is powerful, but messy to read/write. There are multiple ways to express the same logic.
+Once we have a better understanding of the task at hand, we should come up with some basic guidelines for
+formatting and structuring.
+
+### Dependencies between variables
+
+We favor a relatively simple logic and no automatic dependency resolution/re-ordering of questions or variables. 
+If things are ordered incorrectly (e.g. the skipIf in a question refers a future question), the questionaire author should discover and repair the issue during testing.
+
+### Answer Representation
+
+For each answered question, we multiple JSON-Logic variables prefixed with the question id:
+
+```
+question_id.value // string, array of string, number, boolean, date, depending on type
```

Comment:
I would favour something like
Question[]
where each question is defined by
```json
{
  id: "4534534-56543fg-435dfg-43545",  
  answer: {
    type: "Boolean",
    value: true
  },
}
```

Reply:
Since the answer should be addressable by its value, we decided not to have an additional id. For FHIR QuestionnaireResponse we can create an ID on the fly with `value` and `question.id`.

Reply:
Ok, so this is only the internal representation while filling out the form?

Reply:
This is the representation that get's injected into the JSON logic evaluation process, e.g. it defines which values about answers are available and how they are addressed.

The idea was to inject an object for each given answer with a set of properties used for evaluation.

I'm sure this object will grow, but it's also super important that answers to different questions are easy to address.